frb_codegen/src/commands.rs

```diff
@@ -274,3 +283,88 @@ pub fn build_runner(dart_root: &str) -> Result {
     }
     Ok(())
 }
+
+pub fn guess_context(dart_root: &str) -> std::result::Result<Context, Error> {
+    info!("Guessing context the runner is run into");
+    let pubspec = PathBuf::from(dart_root).join("pubspec.lock");
+    if !pubspec.exists() {
+        return Err(Error::StringError(format!(
+            "missing pubspec.lock in {}",
+            dart_root
+        )));
+    }
+    let pubspec = std::fs::read_to_string(pubspec)
+        .map_err(|_| Error::StringError(format!("unable to read pubspec.lock in {}", dart_root)))?;
+    let pubspec: Pubspec = serde_yaml::from_str(&pubspec).map_err(|_| {
+        Error::StringError(format!("unable to parse pubspec.lock in {}", dart_root))
+    })?;
+    if pubspec.packages.contains_key("flutter") {
+        return Ok(Context::Flutter);
+    }
+    Ok(Context::Dart)
+}
+
+#[derive(Debug, PartialEq)]
+pub enum Context {
```

Comment:
Well I am considering the naming... "Context" does not seem to be very intuitive I guess...

Reply:
Toolchain ?

Reply:
maybe DartToolchain? (since toolchain can mean rust toolchain etc)

Reply:
if you're not annoyed by `DartToolchain::Dart` repetition, then I'm fine I'll make the changes.

Reply:
That sounds good to me :)

Reply:
Ok done in 4c7f1d46b74c858ef167bebae26f13e852240a1e